### ISSUE TITLE
Incompatible with Apple M1 (workaround provided)

### ISSUE THREAD

**Issue by bt-nia** [yeswehack/ywh2bugtracker#7]
Hi all!

_tl;dr: workaround is to run it with rosetta: spawn a shell in intel mode with `arch -x86_64 zsh`, verify `i386` is set by running `arch` again, then just `cd` into the git folder and `run poetry install` and then `poetry run ywh2bt-gui` (this will take some time, but the window will appear._ 

This project is unfortunately incompatible with M1 based devices. The reason for that is `pyside2`, which does not have M1 support. 

Usually replacing `pyside2` with `pyside2` (version `6.2.2` up) is enough, but one component that this code base depends on seems to not have been ported to version 6 yet. 
Anyway, here are the steps I've taken so far until I hit a complete road block (for anyone who wants to try to migrate):
Change `/pyproject.toml`: `PySide2 = "^5.15.1"` to `PySide2 = ">6.2.2"`

Run `find ./ywh2bt/gui -name \*.py -exec sed -i'' -e 's/PySide2/PySide6/g' {} \;` to replace all old references

Change import in `ywh2bugtracker/ywh2bt/gui/widgets/main_window.py` of `QAction` from `PySide2.QtWidgets` to from `PySide6.QtGui`

Now, it fails with: `ImportError: cannot import name 'QWebEnginePage' from 'PySide6.QtWebEngineWidgets'`
From what I found, this import has not been ported yet.

**Comment by m-honel-ywh2bt** [comment]
Hi,

we've finaly found some time to fix some issues in this project.

As of version 2.8.0 released today (2024-01-11), the switch has been made to PySide6 and Apple M1 is now supported for python >=3.7,<=3.12.

The GUI has also been made optional so don't forget to `pip install "ywh2bt[gui]" --upgrade` otherwise PySide and other dependencies won't be installed.

Thank you for your initial research on this issue and thanks for your patience.
